ServiceProviderExtensionStorageTest will fail under jdk 14

When building pf4j project under jdk 14, org.pf4j.processor.ServiceProviderExtensionStorageTest will fail due to Mockito's mocking failure.
```
[INFO] Running org.pf4j.processor.ServiceProviderExtensionStorageTest
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.003 s <<< FAILURE! - in org.pf4j.processor.ServiceProviderExtensionStorageTest
[ERROR] ensureReadingExtensionsProducesCorrectListOfExtensions  Time elapsed: 0.001 s  <<< ERROR!
org.mockito.exceptions.base.MockitoException:

Mockito cannot mock this class: interface javax.annotation.processing.Filer.

Mockito can only mock non-private & non-final classes.
If you're not sure why you're getting this error, please report to the mailing list.


Java               : 14
JVM vendor name    : Oracle Corporation
JVM vendor version : 14.0.2+12-46
JVM name           : OpenJDK 64-Bit Server VM
JVM version        : 14.0.2+12-46
JVM info           : mixed mode, sharing
OS name            : Windows 10
OS version         : 10.0


Underlying exception : java.lang.IllegalArgumentException: org.mockito.codegenFiler$MockitoMock$1802809777 must be defined in the same package as org.mockito.codegen.InjectionBase/package
        at org.pf4j.processor.ServiceProviderExtensionStorageTest.ensureReadingExtensionsProducesCorrectListOfExtensions(ServiceProviderExtensionStorageTest.java:82)
Caused by: java.lang.IllegalArgumentException: org.mockito.codegenFiler$MockitoMock$1802809777 must be defined in the same package as org.mockito.codegen.InjectionBase/package
        at org.pf4j.processor.ServiceProviderExtensionStorageTest.ensureReadingExtensionsProducesCorrectListOfExtensions(ServiceProviderExtensionStorageTest.java:82)
```

--- a/pom.xml
+++ b/pom.xml
@@ -51,7 +51,7 @@
 
         <junit.version>5.4.0</junit.version>
         <hamcrest.version>2.1</hamcrest.version>
-        <mockito.version>2.24.0</mockito.version>
+        <mockito.version>3.8.0</mockito.version>
         <cobertura.version>2.7</cobertura.version>
         <coveralls.version>3.1.0</coveralls.version>
 

